File: ClientApp/src/app/models/competition/add-league.model.ts
import { AddLeagueTeamModel } from './add-league-team.model';

export class AddLeagueModel {
    competitionStepId: number;
    teamCount: number;
    isHomeAway: boolean;
    peerToPeerPlayCount: number;
    riseTeamCount: number;
    fallTeamCount: number;
    wonPoint: number;
    drawPoint: number;
    lostPoint: number;
    leagueTeams: AddLeagueTeamModel[];
}

File: Models/Competition/AddLeagueModel.cs
using System.Collections.Generic;

namespace champi.Models.Competition
{
    public class AddLeagueModel
    {
        public long CompetitionStepId { get; set; }
        public int TeamCount { get; set; }
        public bool IsHomeAway { get; set; }
        public int PeerToPeerPlayCount { get; set; }
        public int RiseTeamCount { get; set; }
        public int FallTeamCount { get; set; }
        public int WonPoint { get; set; }
        public int DrawPoint { get; set; }
        public int LostPoint { get; set; }
        public List<AddLeagueTeamModel> LeagueTeams { get; set; }
    }
}